### ISSUE TITLE
qwe

### ISSUE THREAD

**Issue by duguodong7** [duguodong7/duguodong7.github.io#1]
class ResBlock_LCN(nn.Module):
    def __init__(
        self, conv, n_feats, kernel_size,
        bias=True, LCN=False, act=nn.ReLU(True), res_scale=1, group=1, dilation=[6,6]):
        super(ResBlock, self).__init__()

        m = []
        for i in range(2):
            m.append(conv(n_feats, n_feats, kernel_size, bias=bias))
            if LCN: m.append(LCN(group, dilation, n_feats)) 
            if i == 0: m.append(act)

        self.body = nn.Sequential(*m)
        self.res_scale = res_scale

    def forward(self, x):
        res = self.body(x).mul(self.res_scale)
        res += x

        return res
    

class LCN(nn.Module):
    def __init__(self, group, dilation):
        super(ResBlock, self).__init__()
        self.group = group
        self.dilation = dilation
        self.affine = nn.Linear(1, 1, bias=True)

    def forward(self, x):
        N, C = x.size()[0], x.size()[1]

        kernel = torch.tensor([[1,-1],[-1,1]], dtype=torch.float32).view(1,1,2,2)
        kernel = torch.cat([kernel]*N,0)
        kernel = torch.cat([kernel]*C/self.group, 1)

        intergral_x = torch.cumsum(torch.cumsum(x, dim=4), dim=3)
        S = F.conv2d(intergral_x, weight=kernel, group=self.group, dilation=self.dilation, padding=self.dilation/2)

        x_sq = torch.pow(x, 2)
        intergral_x_sq = torch.cumsum(torch.cumsum(x_sq, dim=4), dim=3)
        S_sq = F.conv2d(intergral_x_sq, weight=kernel, group=self.group, dilation=self.dilation, padding=self.dilation/2)

        miu = S/n
        sigma = torch.sqrt((S_sq/n - torch.pow(S/n, 2)) + 1e-8)
        y = self.affine((x-miu)/sigma)

        return y

**Comment by duguodong7** [comment]
class CapsLayer(nn.Module):
    def __init__(self, n_feats=64, n_blks=4, scale=3):
        super(ConvLayer, self).__init__()
        self.n_blks = n_blks
        self.scale = scale
        s = np.power(2, self.n_blks)
        m = [ResBlock(n_feats) for _ in range(n_blks)]
        self.layer = nn.Sequential(*m)

        self.upslayer = nn.Conv2d(n_feats*s*s, n_feats*scale*scale, 3, padding=1, groups=n_feats, bias=True)

    def forward(self, x):
    	N, C, H, W = x.size(0), x.size(1), x.size(2), x.size(3)
        out = self.layer(x)
        out = out.view(N, C, H, s, W, s).permute(0,1,3,5,2,4).contiguous()
        out = out.view(N, C*s*s, H, W)
        up = self.upslayer(out)
        # out = out.view(N, C*s*s, H, W)
        return out
